rfc-016-slack-channel-names.md

```diff
@@ -0,0 +1,21 @@
+# Slack channel names
+
+## Summary
+
+Guidelines for slack channel naming conventions building on [Slack's own guidelines](https://get.slack.help/hc/en-us/articles/217626408-Create-guidelines-for-channel-names).
+
+## Problem
+
+There is no standard naming convention for Slack channel names and as we grow they become harder to discover. It's difficult to understand whether a channel is for discussion on topics or with teams. For example some SRE questions go into #live-services, some into #help, #engineering and even into #announcements.
+
+## Proposal
+
+- You SHOULD create discussion channels for particular engineering topics such as Kubernetes, React, etc.
+- You SHOULD create team channels for all teams within the business whether delivery teams, sales, etc.
+- You SHOULD create project channels for particular team projects that have a focussed goal, an example would be Homes HIF is a project of the Homes team.
+- You MAY create collaboration channels and invite customers to them if you want to use Slack as a communication tool.
+
+- You MUST prefix engineering topics with `#eng-`. For example, `#eng-k8s`, `#eng-react`, `#eng-tdd`.
```

Comment:
If all channels have a prefix I argue that the channel list will be less readable as you have to skip past all the prefixes to find the topic.

i.e 
- #eng-thing
- #eng-other-thing
- #eng-thing-you-want-to-find
- #eng-other-thing

as opposed to 
- #thing
- #other-thing
- #thing-you-want-to-find
- #other-thing

I am in agreement with using keywords but I would like the ability to weave them into the name
i.e #hackney-dev-team or #better-eng

**Note:** There is a field called `Purpose` in slack, granted its hidden in the channel about

Reply:
Yeah I've based the proposal on [Slack's own guidelines](https://get.slack.help/hc/en-us/articles/217626408-Create-guidelines-for-channel-names). For me in my list of slack channels it would be neater to group them by type so if I'm looking for eng topics they are grouped together. If I'm looking for teams they are grouped by that.

This may well be less of a problem for people in fewer channels.

Reply:
I share @cob16's concerns re readability. especially on mobile. I think as long as the channel names are descriptive enough it's not difficult to find what you're looking for.

Reply:
This is a slack recommended approach so shouldn't be too bad.

Shouldn't reduce mobile discoverability, in fact will make it easier to discover engineering topics for example.